### PR TITLE
Add companion app

### PR DESCRIPTION
**Pull request by xrigau**
Add a companion app that can be used to pause and resume the game (not fully working) and also update & display the current gamer tag

### CODE REVIEW

File: dungeon-crawler/android_mobile/src/main/java/com/novoda/dungeoncrawler/MainActivity.java [hunk 1]
```diff
@@ -40,10 +50,23 @@ protected void onResume() {
         Log.d("TUT", "Game starting");
     }
 
+    @Override
+    public void onPauseGame() {
+        Toast.makeText(this, "Game Paused", Toast.LENGTH_SHORT).show();
+        game.onPauseGame();
```
Comment (blundell) [inline dungeon-crawler/android_mobile/src/main/java/com/novoda/dungeoncrawler/MainActivity.java]: I think this should be `game.pause()` you want to tell it to pause, not inform it that the game has been paused? (which makees less sense right?)

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/GamePauseObservable.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.novoda.dungeoncrawler;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.ValueEventListener;
+
+class GamePauseObservable {
```
Comment (blundell) [inline dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/GamePauseObservable.java]: should this have firebase or remote in the name?

File: dungeon-crawler/companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import android.support.v4.app.Fragment
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import kotlinx.android.synthetic.main.fragment_game.*
+
+class GameFragment : Fragment() {
```
Comment (blundell) [inline dungeon-crawler/companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt]: hello kotline fraggos

**Comment by xrigau** [comment]
The pause button is hidden for now

File: dungeon-crawler/android_mobile/src/main/java/com/novoda/dungeoncrawler/MainActivity.java (hunk 1 above)
Comment (xrigau) [reply dungeon-crawler/android_mobile/src/main/java/com/novoda/dungeoncrawler/MainActivity.java]: Done in aba87dcbcf9576d27a95a4eef5d2011f5525ee53

File: dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/GamePauseObservable.java (hunk 2 above)
Comment (xrigau) [reply dungeon-crawler/android/src/main/java/com/novoda/dungeoncrawler/GamePauseObservable.java]: Done in 0f611b7100f1d60cf249885232b58d7c4e43d5f8

File: dungeon-crawler/companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt (hunk 3 above)
Comment (xrigau) [reply dungeon-crawler/companion/src/main/java/com/novoda/dungeoncrawlercompanion/GameFragment.kt]: 😏

**Pull request merged**